### PR TITLE
Dollar sign in snippets must not be considered as a capture instruction

### PR DESCRIPTION
**Pull request by lcacheux**
I had this bug when snippet contained strings with $.

If $ is followed by a number, the Javadoc is not properly formatted.
If $ is followed by a space, the Javadoc command throws an exception : 

`javadoc: error - In doclet class org.apidesign.javadoc.codesnippet.Doclet,  method start has thrown an exception java.lang.reflect.InvocationTargetException
java.lang.IllegalArgumentException: Illegal group reference
        at java.util.regex.Matcher.appendReplacement(Matcher.java:857)
        at org.apidesign.javadoc.codesnippet.Snippets.boldJavaKeywords(Snippets.java:489)
        at org.apidesign.javadoc.codesnippet.Snippets$Item.toString(Snippets.java:651)
        at org.apidesign.javadoc.codesnippet.Snippets$1.visitFile(Snippets.java:262)
        at org.apidesign.javadoc.codesnippet.Snippets$1.visitFile(Snippets.java:216)
        at java.nio.file.Files.walkFileTree(Files.java:2670)
        at java.nio.file.Files.walkFileTree(Files.java:2742)
        at org.apidesign.javadoc.codesnippet.Snippets.scanDir(Snippets.java:216)
        at org.apidesign.javadoc.codesnippet.Snippets.findSnippet(Snippets.java:190)
        at org.apidesign.javadoc.codesnippet.Snippets.fixCodesnippets(Snippets.java:84)
        at org.apidesign.javadoc.codesnippet.Doclet.start(Doclet.java:52)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.sun.tools.javadoc.DocletInvoker.invoke(DocletInvoker.java:310)
        at com.sun.tools.javadoc.DocletInvoker.start(DocletInvoker.java:189)
        at com.sun.tools.javadoc.Start.parseAndExecute(Start.java:366)
        at com.sun.tools.javadoc.Start.begin(Start.java:219)
        at com.sun.tools.javadoc.Start.begin(Start.java:205)
        at com.sun.tools.javadoc.Main.execute(Main.java:64)
        at com.sun.tools.javadoc.Main.main(Main.java:54)
1 error
`

### CODE REVIEW

**Comment by jtulach** [comment]
It'd be to have test demonstrating what was broken before the d4c3083 fix. It should be easy to modify [one of the files](https://github.com/jtulach/codesnippet4javadoc/tree/master/testing/src/main/java/org/apidesign/javadoc/testing) to use the `$`.

Can you provide such test, please?

File: testing/src/main/java/org/apidesign/javadoc/testing/SpecialCharacters.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * {@codesnippet dollarSnippet1}
+ * {@codesnippet dollarSnippet2}
+ */
+public class SpecialCharacters {
+    // BEGIN: dollarSnippet1
+    public static final String MY_MONEY = "$100.00";
+    // END: dollarSnippet1
+
+    // BEGIN: dollarSnippet2
+    public static final String MONEY_SIGNS = "$ € £";
```
Comment (jtulach) [inline testing/src/main/java/org/apidesign/javadoc/testing/SpecialCharacters.java]: A test that worths a lot!

**Comment by jtulach** [comment]
Thanks a lot for your fixes (this one and #10). I've prepared new bits at https://maven.java.net/content/repositories/orgapidesign-1077/ - if they work for you I can publish them to Maven central with a browser click.

**Comment by lcacheux** [comment]
Thanks for your merge. I tested again with your repository and it fixed all my issues (before this, I worked with my own custom build).

**Pull request merged**